### PR TITLE
refactor to work with `express-view`

### PR DESCRIPTION
**Pull request by caridy**
- the plugin is now a class definition that can be extended easily.
- leverage `Y.Template.register()` and `Y.Template.get()`, available in yui@3.12.x.
- a compiled version of the template is provisioned under the bundle object for runtime on the server side.
- yui output format is now opt-in.
- example does not use `express-yui` anymore, instead it uses `express-view`.
- the `handlebars` component can be controlled from outside by providing a custom version of it when creating an instance of the plugin.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1 +1,7 @@
-module.exports = require('./lib/');
+/*
+ * Copyright (c) 2013, Yahoo! Inc.  All rights reserved.
+ * Copyrights licensed under the New BSD License.
+ * See the accompanying LICENSE file for terms.
+ */
+
+module.exports = require('./lib/plugin');
```
Comment (isao) [inline index.js]: I know this is consistent with other pkgs, but one could either put the plugin code in this file, or use package.json `"main": "lib/plugin.js" and save a file..

**Comment by isao** [comment]
+1

**Comment by caridy** [comment]
yeah, mostly for consistency! but you are right, in the past that consistency was dictated by the fact that we want to expose multiple implementations of the plugin (yui, amd, es6) that were suppose to use the exact same core, so index was meant boilerplate that. This is not longer the case, so let's move on and use the `main` definition. I will make the changes.

**Comment by caridy** [comment]
- locator-handlebars@0.3.0

**Pull request merged**